File: markdowner-core/src/test/java/org/jesperancinha/parser/markdowner/helper/TagHelperTest.java
package org.jesperancinha.parser.markdowner.helper;

import org.junit.jupiter.api.Test;

import static org.assertj.core.api.Assertions.assertThat;

public class TagHelperTest {

    @Test
    public void testSanitizeTag() {
        final String result = TagHelper.sanitizeTag("### Title1 ##");
        assertThat(result).isEqualTo("Title1 ##");
    }

    @Test
    public void testSanitizeTagAllHashes() {
        final String result = TagHelper.sanitizeTag("#####");
        assertThat(result).isEmpty();
    }

    @Test
    public void testCounHashTags() {
        final int result = TagHelper.counHashTags("### okoko #");
        assertThat(result).isEqualTo(3);
    }
}

File: markdowner-core/src/test/kotlin/org/jesperancinha/parser/markdowner/helper/TagHelperTest.kt
package org.jesperancinha.parser.markdowner.helper

import io.kotest.matchers.shouldBe
import io.kotest.matchers.string.shouldBeEmpty
import org.junit.jupiter.api.Test

class TagHelperTest {
    @Test
    fun `should sanitize tag`() {
        val result = TagHelper.sanitizeTag("### Title1 ##")
        result shouldBe "Title1 ##"
    }

    @Test
    fun `should sanitize all hashes to empty`() {
        val result = TagHelper.sanitizeTag("#####")
        result.shouldBeEmpty()
    }

    @Test
    fun `should count hashtags correctly 3`() {
        val result = TagHelper.counHashTags("### okoko #")
        result shouldBe 3
    }
}